cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -767,3 +765,111 @@ func convertToStackParams(parameters azure.ArmParameters) map[string]*armdeploym
 	}
 	return stackParams
 }
+
+func (d *StackDeployments) ValidatePreflightToResourceGroup(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+	options map[string]any,
+) error {
+	client, err := d.createClient(ctx, subscriptionId)
+	if err != nil {
+		return err
+	}
+
+	stack, err := d.stackFromArmForResourceGroup(ctx, subscriptionId, armTemplate, parameters, tags, options)
+	if err != nil {
+		return err
+	}
+
+	var rawResponse *http.Response
+	ctxWithResp := runtime.WithCaptureResponse(ctx, &rawResponse)
+
+	validateResult, err := client.BeginValidateStackAtResourceGroup(ctxWithResp, resourceGroup, deploymentName, stack, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "resource group")
+	}
+	_, err = validateResult.PollUntilDone(ctx, nil)
```

Comment:
Not very familiar with contexts in go - how come we're using `ctx` here instead of `ctxWithResp`?

Reply:
🙂 Great inquiry @JeffreyCA ! I suspect we may need to use `ctxWithResp` for this to be handled correctly.

go [context](https://pkg.go.dev/context) are immutable data structures (for concurrency reasons -- the main use is to signal cancellation and deadlines/timeouts across goroutines), so with each `WithContext`, you get back a new context with added values in the context property bag.

I think it's perfectly okay for us to shadow the `ctx` on line 790 here, i.e. `ctx = policy.WithCaptureReseponse(ctx, &rawResponse)` works perfectly fine in my mind.

Reply:
I see, thank you both for clarifying! 🙂